druid/examples/open_save.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use druid::widget::{Align, Button, Flex, TextBox};
+use druid::{
+    AppDelegate, AppLauncher, Command, DelegateCtx, Env, FileDialogOptions, FileInfo, FileSpec,
+    LocalizedString, Target, Widget, WindowDesc,
+};
+
+struct Delegate;
+
+pub fn main() {
+    let main_window = WindowDesc::new(ui_builder)
+        .title(LocalizedString::new("open-save-demo").with_placeholder("Opening/Saving Demo"));
+    let data = "Type here.".to_owned();
+    AppLauncher::with_window(main_window)
+        .delegate(Delegate)
+        .use_simple_logger()
+        .launch(data)
+        .expect("launch failed");
+}
+
+fn ui_builder() -> impl Widget<String> {
+    let txt = FileSpec::new("Text file", &["txt"]);
+    let other = FileSpec::new("Bogus file", &["foo", "bar", "baz"]);
+    let save_dialog_options = FileDialogOptions::new()
+        .allowed_types(vec![txt, other])
+        .default_type(txt);
```

Comment:
Let's reverse the vector to `[other, txt]` so that this sample can show the default type working. Does it work for you? On my Ubuntu 19.10 it doesn't seem to actually choose the text file if `other` is first in the vector.

Reply:
Indeed, it wasn't working with the other order. I've added some extra checks in dialog.rs, and now all the combinations I've tried seem to work.